crates/reward/rpc/runtime-api/src/lib.rs

```diff
@@ -7,16 +7,24 @@ use frame_support::dispatch::DispatchError;
 use oracle_rpc_runtime_api::BalanceWrapper;
 
 sp_api::decl_runtime_apis! {
-    pub trait RewardApi<AccountId, VaultId, CurrencyId, Balance> where
-    AccountId: Codec,
-    VaultId: Codec,
+    pub trait RewardApi<AccountId, VaultId, CurrencyId, Balance, BlockNumber, UnsignedFixedPoint> where
+        AccountId: Codec,
+        VaultId: Codec,
         CurrencyId: Codec,
-        Balance: Codec
+        Balance: Codec,
+        BlockNumber: Codec,
+        UnsignedFixedPoint: Codec,
     {
         /// Get a given user's rewards due
         fn compute_escrow_reward(account_id: AccountId, currency_id: CurrencyId) -> Result<BalanceWrapper<Balance>, DispatchError>;
 
         /// Get a given vault's rewards due
         fn compute_vault_reward(vault_id: VaultId, currency_id: CurrencyId) -> Result<BalanceWrapper<Balance>, DispatchError>;
+
+        /// Estimate staking reward rate for a period of N blocks
+        fn estimate_escrow_reward_rate(account_id: AccountId, amount: Option<Balance>, lock_time: Option<BlockNumber>, period: BlockNumber) -> Result<UnsignedFixedPoint, DispatchError>;
+
+        /// Estimate vault reward rate for a period of N blocks
```

Comment:
Is it a rate (i.e. reward per block), or the total rewards of `N` blocks? Please document clearly

Reply:
It is the reward rate for a period of N blocks

Reply:
The wording was confusing to me but nvm, after looking at the code I see what it does. The expected use case will be to have N=1year, s.t. the return value is the apr

Reply:
I think N can sometimes be 7 days, but in our UI I think it is only ever 1 year so happy to have that change. @bvotteler do you agree?

Reply:
> I think N can sometimes be 7 days, but in our UI I think it is only ever 1 year so happy to have that change. @bvotteler do you agree?

1 year is fine, we can scale up and down accordingly on the consumer side.
Querying for N blocks obviously gives us more flexibility in queries, but definitely not a deal breaker without that.